src/lib.rs

```diff
@@ -532,6 +532,80 @@ impl Discord {
 		check_empty(request!(self, delete, "/channels/{}/permissions/{}", channel, id))
 	}
 
+	/// Add a `Reaction` to a `Message`.
+	/// 
+	/// # Examples
+	/// Add a twemoji to a `Message`:
+	///
+	/// ```ignore
+	/// // Assuming that a `Discord` instance, channel, message have
+	/// // already been previously defined.
+	/// let _ = discord.add_reaction(&channel.id, message.id, "👌", None);
+	/// ```
+	/// 
+	/// Add a custom emoji to a `Message`:
+	/// ```ignore
+	/// // Assuming that a `Discord` instance, channel, message have
+	/// // already been previously defined.
+	/// use discord::model::EmojiId;
+	///
+	/// let _ = discord.add_reaction(&channel.id, message.id, "ThisIsFine", Some(EmojiId(236147133370204161)));
+	/// ```
+	///
+	/// Requires the `ADD_REACTIONS` permission.
+	pub fn add_reaction(&self, channel: &ChannelId, message: MessageId, emoji_name: &str, emoji_id: Option<EmojiId>) -> Result<()> {
+		let emoji = match emoji_id {
+			// it's a custom emoji
+			Some(id) => format!("{}:{}", emoji_name, id.0),
+			// it's a twemoji
+			None => emoji_name.to_string(),
+		};
+		check_empty(request!(self, put, "/channels/{}/messages/{}/reactions/{}/@me", channel, message, emoji))
+	}
+
+	/// Delete a `Reaction` from a `Message`.
+	/// 
+	/// # Examples
+	/// Delete a twemoji from your `Message`:
+	///
+	/// ```ignore
+	/// // Assuming that a `Discord` instance, channel, message, state have
+	/// // already been previously defined.
+	/// let _ = discord.delete_reaction(&channel.id, message.id, None, "👌", None);
+	/// ```
+	/// 
+	/// Delete a custom emoji from your `Message`:
+	/// ```ignore
+	/// // Assuming that a `Discord` instance, channel, message have
+	/// // already been previously defined.
+	/// use discord::model::EmojiId;
+	///
+	/// let _ = discord.delete_reaction(&channel.id, message.id, None, "ThisIsFine", Some(EmojiId(236147133370204161)));
+	/// ```
+	pub fn delete_reaction(&self, channel: &ChannelId, message: MessageId, user_id: Option<UserId>, emoji_name: &str, emoji_id: Option<EmojiId>) -> Result<()> {
```

Comment:
Not sure, haven't actually tried this since i piggybacked of off discord.js's implementation, though i'm going to talk with them if it is possible.